tests/cops_ui/test_cops_e2e.py

```diff
@@ -0,0 +1,117 @@
+import requests
+from tests import markers
+from selenium.webdriver.common.keys import Keys
+
+import pytest
+
+import json
+import urllib
+from urllib.request import urlopen
+
+from urllib.error import HTTPError, URLError
+from requests.exceptions import ConnectionError
+
+from time import sleep
+
+
+"""
+End to end test of cops(-staging).openstax.org
+Creates jobs for 2 collections, verifies that they were successfully executed and pdf is created
+Latest update on 26/02/2020
+"""
+
+
+@markers.parametrize(
+    "colid, style, bserver", [("col11992", "astronomy", "qa"), ("col11496", "anatomy", "staging")]
+)
+def test_cops_ui(selenium, cops_base_url, create_new_pdf_job, create_button, colid, style, bserver):
+
+    # GIVEN a cops base URL
+    # WHEN making a request to cops
+
+    try:
+
+        # verifying that cops URL is active
+        response = requests.get(cops_base_url)
+
+    except HTTPError:
+        pytest.exit("<<<<<<<<<< HTTP Error: COPS URL IS INVALID OR UNREACHABLE >>>>>>>>>>")
+    except URLError:
+        pytest.exit("<<<<<<<<<< URL Error: COPS URL IS INVALID OR UNREACHABLE >>>>>>>>>>")
+    except ConnectionError:
+        pytest.exit("<<<<<<<<<< Connection Error: COPS URL IS INVALID OR UNREACHABLE >>>>>>>>>>")
+
+    else:
+
+        # opening 'CREATE A NEW PDF JOB' dialog
+        selenium.get(cops_base_url)
+        new_pdf = selenium.find_element_by_xpath(create_new_pdf_job)
+        new_pdf.click()
+
+        # tabbing through fields and inputting colID, style and server
+        selenium.find_element_by_tag_name("body").send_keys(Keys.TAB, colid)
+        selenium.find_element_by_tag_name("body").send_keys(Keys.TAB)
+        selenium.find_element_by_tag_name("body").send_keys(Keys.TAB, style)
+        selenium.find_element_by_tag_name("body").send_keys(Keys.TAB, bserver)
+
+        # clicking 'CREATE' button to start the jobs
+        create_button_xp = selenium.find_element_by_xpath(create_button)
+        create_button_xp.click()
+        create_button_xp.click()
+
+        # THEN we should get successful connection
+        assert response.status_code == 200
+
+
+def test_job_results(selenium, cops_base_url, cops_api_url):
+
+    sleep(2)
+
+    while True:
```

Comment:
Will this spin forever potentially if any of the jobs fail and never reach `completed`, or does the test harness kill it after some point? If the former, perhaps we can give up after so many iterations (e.g. if we want to give up after X minutes, give up after X / sleep_time).

Reply:
Yes. You are right. I need to add some kind of timeout. Have to find a good solution.

Reply:
@omehes - I took a look at the timeout mechanism in this commit and it looks like it should address the issue  :+1: